docs/nddata/covariance.rst

```diff
@@ -0,0 +1,604 @@
+...         + np.diag(np.full(npts-1, 0.5, dtype=float), k=-1)
+...         + np.diag(np.full(npts, 1.0, dtype=float), k=0)
+...         + np.diag(np.full(npts-1, 0.5, dtype=float), k=1)
+...         + np.diag(np.full(npts-2, 0.2, dtype=float), k=2))
+>>> c
+array([[1. , 0.5, 0.2, 0. , 0. , 0. , 0. , 0. , 0. , 0. ],
+       [0.5, 1. , 0.5, 0.2, 0. , 0. , 0. , 0. , 0. , 0. ],
+       [0.2, 0.5, 1. , 0.5, 0.2, 0. , 0. , 0. , 0. , 0. ],
+       [0. , 0.2, 0.5, 1. , 0.5, 0.2, 0. , 0. , 0. , 0. ],
+       [0. , 0. , 0.2, 0.5, 1. , 0.5, 0.2, 0. , 0. , 0. ],
+       [0. , 0. , 0. , 0.2, 0.5, 1. , 0.5, 0.2, 0. , 0. ],
+       [0. , 0. , 0. , 0. , 0.2, 0.5, 1. , 0.5, 0.2, 0. ],
+       [0. , 0. , 0. , 0. , 0. , 0.2, 0.5, 1. , 0.5, 0.2],
+       [0. , 0. , 0. , 0. , 0. , 0. , 0.2, 0.5, 1. , 0.5],
+       [0. , 0. , 0. , 0. , 0. , 0. , 0. , 0.2, 0.5, 1. ]])
+
+In this case, the correlation matrix and the covariance matrix are *identical*
+because the elements of the variance vector are all unity.
+
+With a correlation matrix, we can construct the covariance matrix with any
+arbitrary variance vector.  Continuing the example above, the following creates
+a new covariance matrix with a variance vector with all elements equal to 4:
+
+>>> new_var = np.full(npts, 4., dtype=float)
+>>> new_c = c * np.sqrt(new_var[:,None] * new_var[None,:])
+>>> new_c
+array([[4. , 2. , 0.8, 0. , 0. , 0. , 0. , 0. , 0. , 0. ],
+       [2. , 4. , 2. , 0.8, 0. , 0. , 0. , 0. , 0. , 0. ],
+       [0.8, 2. , 4. , 2. , 0.8, 0. , 0. , 0. , 0. , 0. ],
+       [0. , 0.8, 2. , 4. , 2. , 0.8, 0. , 0. , 0. , 0. ],
+       [0. , 0. , 0.8, 2. , 4. , 2. , 0.8, 0. , 0. , 0. ],
+       [0. , 0. , 0. , 0.8, 2. , 4. , 2. , 0.8, 0. , 0. ],
+       [0. , 0. , 0. , 0. , 0.8, 2. , 4. , 2. , 0.8, 0. ],
+       [0. , 0. , 0. , 0. , 0. , 0.8, 2. , 4. , 2. , 0.8],
+       [0. , 0. , 0. , 0. , 0. , 0. , 0.8, 2. , 4. , 2. ],
+       [0. , 0. , 0. , 0. , 0. , 0. , 0. , 0.8, 2. , 4. ]])
+
+Or likewise for heteroscedastic data:
+
+>>> new_var = (1. + np.absolute(np.arange(npts) - npts//2).astype(float))**2
+>>> new_var
+array([36., 25., 16.,  9.,  4.,  1.,  4.,  9., 16., 25.])
+>>> new_c = c * np.sqrt(new_var[:,None] * new_var[None,:])
+>>> new_c
+array([[36. , 15. ,  4.8,  0. ,  0. ,  0. ,  0. ,  0. ,  0. ,  0. ],
+       [15. , 25. , 10. ,  3. ,  0. ,  0. ,  0. ,  0. ,  0. ,  0. ],
+       [ 4.8, 10. , 16. ,  6. ,  1.6,  0. ,  0. ,  0. ,  0. ,  0. ],
+       [ 0. ,  3. ,  6. ,  9. ,  3. ,  0.6,  0. ,  0. ,  0. ,  0. ],
+       [ 0. ,  0. ,  1.6,  3. ,  4. ,  1. ,  0.8,  0. ,  0. ,  0. ],
+       [ 0. ,  0. ,  0. ,  0.6,  1. ,  1. ,  1. ,  0.6,  0. ,  0. ],
+       [ 0. ,  0. ,  0. ,  0. ,  0.8,  1. ,  4. ,  3. ,  1.6,  0. ],
+       [ 0. ,  0. ,  0. ,  0. ,  0. ,  0.6,  3. ,  9. ,  6. ,  3. ],
+       [ 0. ,  0. ,  0. ,  0. ,  0. ,  0. ,  1.6,  6. , 16. , 10. ],
+       [ 0. ,  0. ,  0. ,  0. ,  0. ,  0. ,  0. ,  3. , 10. , 25. ]])
+
+Reordering and Subsets
```

Comment:
I'm not quite sure where the right place in this section is to mention it but I think it is worth mentioning that the `Covariance` class has methods for doing these operations so that one does not have to do these (impressive) numpy gymnastics.

Reply:
I added short note sections for each of these three sections pointing to the "utility" function descriptions at the end of the doc.